### ISSUE TITLE
Fresh minion install (2015.8.3) returns service __virtual__ is False errors on salt-calls

### ISSUE THREAD

**Issue by c4t3l** [saltstack/salt#29630]
Hello,

Today I installed salt-minion 2015.8.3 from the saltstack repository and I am seeing the following:
```
root@REDACTED0 ~]# salt-call service.status puppet
[ERROR   ] You should upgrade pyOpenSSL to at least 0.14.1 to enable the use of X509 extensions
[ERROR   ] Exception raised when processing __virtual__ function for nova. Module will not be loaded global name 'novaclient' is not defined
[WARNING ] salt.loaded.int.module.nova.__virtual__() is wrongly returning `None`. It should either return `True`, `False` or a new name. If you're the developer of the module 'nova', please fix this.
'service' __virtual__ returned False
```
Two days ago I installed the very same packages from the repos and see this:
```
[root@REDACTED1 ~]# salt-call service.status puppet
[INFO    ] Executing command '/sbin/service puppet status' in directory '/root'
local:
    True

Grain info:

REDACTED0:

local:
    ----------
    SSDs:
    biosreleasedate:
        07/09/2012
    biosversion:
        6.00
    cpu_flags:
        - fpu
        - vme
        - de
        - pse
        - tsc
        - msr
        - pae
        - mce
        - cx8
        - apic
        - sep
        - mtrr
        - pge
        - mca
        - cmov
        - pat
        - pse36
        - clflush
        - dts
        - acpi
        - mmx
        - fxsr
        - sse
        - sse2
        - ss
        - ht
        - syscall
        - nx
        - rdtscp
        - lm
        - constant_tsc
        - arch_perfmon
        - pebs
        - bts
        - nopl
        - xtopology
        - tsc_reliable
        - nonstop_tsc
        - aperfmperf
        - pni
        - ssse3
        - cx16
        - sse4_1
        - sse4_2
        - popcnt
        - hypervisor
        - lahf_lm
        - ida
        - dtherm
    cpu_model:
        Intel(R) Xeon(R) CPU           E5540  @ 2.53GHz
    cpuarch:
        x86_64
    domain:
        example.com
    env:
        prod
    fqdn:
        REDACTED0
    fqdn_ip4:
        - XXX.XXX.XXX.XXX
    fqdn_ip6:
    gpus:
        |_
          ----------
          model:
              SVGA II Adapter
          vendor:
              unknown
    host:
        REDACTED0
    hwaddr_interfaces:
        ----------
        eth0:
            00:50:56:a3:1a:ad
        lo:
            00:00:00:00:00:00
    id:
        REDACTED0
    init:
        upstart
    ip4_interfaces:
        ----------
        eth0:
            - XXX.XXX.XXX.XXX
        lo:
            - 127.0.0.1
    ip6_interfaces:
        ----------
        eth0:
            - fe80::250:56ff:fea3:1aad
        lo:
            - ::1
    ip_interfaces:
        ----------
        eth0:
            - XXX.XXX.XXX.XXX
            - fe80::250:56ff:fea3:1aad
        lo:
            - 127.0.0.1
            - ::1
    ipv4:
        - XXX.XXX.XXX.XXX
        - 127.0.0.1
    ipv6:
        - ::1
        - fe80::250:56ff:fea3:1aad
    kernel:
        Linux
    kernelrelease:
        3.8.13-68.el6uek.x86_64
    locale_info:
        ----------
        defaultencoding:
            UTF8
        defaultlanguage:
            en_US
        detectedencoding:
            UTF-8
    localhost:
        REDACTED0
    lsb_distrib_codename:
        Oracle Linux Server 6.7" 
    lsb_distrib_id:
        Oracle Linux Server"
    lsb_distrib_release:
        6.7" 
    machine_id:
        d5b6b8d003a54ec38ff5fdfc0000000b
    manufacturer:
        VMware, Inc.
    master:
        salt
    mdadm:
    mem_total:
        12017
    nodename:
        srvhouplmlts01
    num_cpus:
        4
    num_gpus:
        1
    os:
        OEL
    os_family:
        RedHat
    osarch:
        x86_64
    oscodename:
        Oracle Linux Server 6.7"
    osfinger:
        Oracle Linux Server"-6
    osfullname:
        Oracle Linux Server"
    osmajorrelease:
        6
    osrelease:
        6.7"
    osrelease_info:
        - 6
        - 7"
    path:
        /usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin
    productname:
        VMware Virtual Platform
    ps:
        ps -efH
    pythonexecutable:
        /usr/bin/python2.6
    pythonpath:
        - /usr/bin
        - /usr/lib64/python26.zip
        - /usr/lib64/python2.6
        - /usr/lib64/python2.6/plat-linux2
        - /usr/lib64/python2.6/lib-tk
        - /usr/lib64/python2.6/lib-old
        - /usr/lib64/python2.6/lib-dynload
        - /usr/lib64/python2.6/site-packages
        - /usr/lib64/python2.6/site-packages/gtk-2.0
        - /usr/lib/python2.6/site-packages
    pythonversion:
        - 2
        - 6
        - 6
        - final
        - 0
    saltpath:
        /usr/lib/python2.6/site-packages/salt
    saltversion:
        2015.8.3
    saltversioninfo:
        - 2015
        - 8
        - 3
        - 0
    selinux:
        ----------
        enabled:
            False
        enforced:
            Disabled
    serialnumber:
        VMware-42 23 53 d8 92 f5 9e 96-aa c4 a7 16 bd 61 f1 68
    server_id:
        258726120
    shell:
        /bin/bash
    uuid:
        422353d8-92f5-9e96-aac4-a716bd61f168
    virtual:
        VMWare
    zmqversion:
        4.0.5


REDACTED1:

local:
    ----------
    SSDs:
    biosreleasedate:
        04/14/2014
    biosversion:
        6.00
    cpu_flags:
        - fpu
        - vme
        - de
        - pse
        - tsc
        - msr
        - pae
        - mce
        - cx8
        - apic
        - sep
        - mtrr
        - pge
        - mca
        - cmov
        - pat
        - pse36
        - clflush
        - dts
        - mmx
        - fxsr
        - sse
        - sse2
        - ss
        - syscall
        - nx
        - rdtscp
        - lm
        - constant_tsc
        - arch_perfmon
        - pebs
        - bts
        - nopl
        - xtopology
        - tsc_reliable
        - nonstop_tsc
        - aperfmperf
        - pni
        - pclmulqdq
        - ssse3
        - cx16
        - pcid
        - sse4_1
        - sse4_2
        - x2apic
        - popcnt
        - aes
        - xsave
        - avx
        - f16c
        - rdrand
        - hypervisor
        - lahf_lm
        - ida
        - arat
        - epb
        - pln
        - pts
        - dtherm
        - fsgsbase
        - smep
    cpu_model:
        Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz
    cpuarch:
        x86_64
    domain:
        example.com
    env:
        prod
    fqdn:
        REDACTED1.example.com
    fqdn_ip4:
        - XXX.XXX.XXX.XXX
    fqdn_ip6:
    gpus:
        |_
          ----------
          model:
              SVGA II Adapter
          vendor:
              unknown
    host:
        REDACTED1
    hwaddr_interfaces:
        ----------
        eth0:
            00:50:56:ab:73:2f
        lo:
            00:00:00:00:00:00
    id:
        REDACTED1
    init:
        upstart
    ip4_interfaces:
        ----------
        eth0:
            - XXX.XXX.XXX.XXX
        lo:
            - 127.0.0.1
    ip6_interfaces:
        ----------
        eth0:
        lo:
    ip_interfaces:
        ----------
        eth0:
            - XXX.XXX.XXX.XXX
        lo:
            - 127.0.0.1
    ipv4:
        - XXX.XXX.XXX.XXX
        - 127.0.0.1
    ipv6:
    kernel:
        Linux
    kernelrelease:
        3.8.13-16.2.1.el6uek.x86_64
    locale_info:
        ----------
        defaultencoding:
            UTF8
        defaultlanguage:
            en_US
        detectedencoding:
            UTF-8
    localhost:
        REDACTED1
    machine_id:
        e71b249ffcc525f107102e810000000c
    manufacturer:
        VMware, Inc.
    master:
        salt
    mdadm:
    mem_total:
        17639
    nodename:
        srvgdyplmsdc01
    num_cpus:
        4
    num_gpus:
        1
    os:
        OEL
    os_family:
        RedHat
    osarch:
        x86_64
    oscodename:
    osfinger:
        Oracle Linux Server-6
    osfullname:
        Oracle Linux Server
    osmajorrelease:
        6
    osrelease:
        6.5
    osrelease_info:
        - 6
        - 5
    path:
        /usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin
    productname:
        VMware Virtual Platform
    ps:
        ps -efH
    pythonexecutable:
        /usr/bin/python2.6
    pythonpath:
        - /usr/bin
        - /usr/lib64/python26.zip
        - /usr/lib64/python2.6
        - /usr/lib64/python2.6/plat-linux2
        - /usr/lib64/python2.6/lib-tk
        - /usr/lib64/python2.6/lib-old
        - /usr/lib64/python2.6/lib-dynload
        - /usr/lib64/python2.6/site-packages
        - /usr/lib64/python2.6/site-packages/gtk-2.0
        - /usr/lib/python2.6/site-packages
    pythonversion:
        - 2
        - 6
        - 6
        - final
        - 0
    saltpath:
        /usr/lib/python2.6/site-packages/salt
    saltversion:
        2015.8.3
    saltversioninfo:
        - 2015
        - 8
        - 3
        - 0
    selinux:
        ----------
        enabled:
            False
        enforced:
            Disabled
    serialnumber:
        VMware-42 2b 5f 5e 68 35 99 43-2f b9 7a 87 e9 dd a2 0e
    server_id:
        0000000000
    shell:
        /bin/bash
    uuid:
        422b5f5e-6835-9943-2fb9-7a87e9dda20e
    virtual:
        VMWare
    zmqversion:
        4.0.5
```

I have already attempted cleaning yum cache and re-installing.  Any help would be greatly appreciated.

:)

**Comment by jfindlay** [comment]
@c4t3l, thanks for the report. The lesser complaints are fixed by #29455 and #29632.  The `global name 'novaclient' is not defined` error is a duplicate of #29511.

**Comment by c4t3l** [comment]
@jfindlay, I am a bit confused.  No service module calls to these minions yield anything of value.  Any state that references the service module fails to run.  Is this behavior caused by the nova error??
   
I see the following:

[root@srvxxx ~]# salt-call service.status puppet -l debug
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: srvxxx.example.com
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Initializing new SAuth for ('/etc/salt/pki/minion', 'srvxxx.example.com', 'tcp://xxx.xxx.xxx.xxx:4506')
[DEBUG   ] Generated random reconnect delay between '1000ms' and '11000ms' (4294)
[DEBUG   ] Setting zmq_reconnect_ivl to '4294ms'
[DEBUG   ] Setting zmq_reconnect_ivl_max to '11000ms'
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'srvxxx.example.com', 'tcp://xxx.xxx.xxx.xxx:4506', 'aes')
[DEBUG   ] Initializing new SAuth for ('/etc/salt/pki/minion', 'srvxxx.example.com', 'tcp://xxx.xxx.xxx.xxx:4506')
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'srvxxx.example.com', 'tcp://XXX.XXX.XXX.XXX:4506', 'clear')
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] Error loading module.dockerng: Docker module could not get imported
[DEBUG   ] LazyLoaded config.get
[DEBUG   ] Error loading module.ifttt: IFTTT Secret Key Unavailable, not loading.
[DEBUG   ] Error loading module.victorops: No VictorOps api key found.
[DEBUG   ] Error loading module.glusterfs: glusterfs server is not installed
[DEBUG   ] Error loading module.nacl: libnacl import error, perhaps missing python libnacl package
[ERROR   ] You should upgrade pyOpenSSL to at least 0.14.1 to enable the use of X509 extensions
[ERROR   ] Exception raised when processing __virtual__ function for nova. Module will not be loaded global name 'novaclient' is not defined
[WARNING ] salt.loaded.int.module.nova.__virtual__() is wrongly returning `None`. It should either return `True`, `False` or a new name. If you're the developer of the module 'nova', please fix this.
[DEBUG   ] Error loading module.nova: Exception raised when processing __virtual__ function for nova. Module will not be loaded global name 'novaclient' is not defined
[DEBUG   ] Error loading module.npm: npm execution module could not be loaded because the npm binary could not be located
[DEBUG   ] Error loading module.boto_cloudwatch: The boto_cloudwatch module cannot be loaded: boto libraries are unavailable.
[DEBUG   ] Error loading module.ipmi: No module named pyghmi.ipmi
[DEBUG   ] Could not LazyLoad service.status
'service' __virtual__ returned False

**Comment by jfindlay** [comment]
@c4t3l, it may be.  Are you able to test the fix, #29196?

**Comment by c4t3l** [comment]
@jfindlay, I bypassed all calls to nova on a test node and the error persisted.  I will find another node and do some testing with #29196.
